src/dataset.rs

```diff
@@ -535,6 +537,95 @@ impl Dataset {
         }
         Ok(Transaction::new(self))
     }
+
+    /// Execute a SQL query against the Dataset. It is equivalent to calling
+    /// [`GDALDatasetExecuteSQL`](https://gdal.org/api/raster_c_api.html#_CPPv421GDALDatasetExecuteSQL12GDALDatasetHPKc12OGRGeometryHPKc).
+    /// Returns a `ResultSet`, which can be treated just as any other [`Layer`], or `None` if the
+    /// query does not have a `ResultSet`, such as `ALTER TABLE`, etc. Note that this is different
+    /// than a query which has an *empty* `ResultSet`.
```

Comment:
To make sure I understand:

Something like `Select * where false` would return an empty `Some(result_set)` where `result_set.len() == 0` whereas `Alter Table ...` would return `None` right?


At a minimum, I have a grammar nit: "different than a query" should be "different __from__ a query" or "distinct from a query"

Alternatively, maybe this phrasing is clearer:

> Returns a `ResultSet`, which can be treated just as any other [`Layer`]. 
>
> Queries, such as `ALTER TABLE`, etc. have no ResultSet, and return `None`, which is distinct from an empty `ResultSet`.

Reply:
This is correct. I've made the suggested changes to the wording.